lib/ansible/modules/cloud/vmware/vmware_datastore_facts.py

```diff
@@ -275,15 +275,7 @@ def main():
 
     result['datastores'] = datastores
 
-    # found a datastore
-    if datastores:
-        module.exit_json(**result)
-    else:
-        msg = "Unable to gather datastore facts"
-        if module.params['name']:
-            msg += " for %(name)s" % module.params
-        msg += " in datacenter %(datacenter)s" % module.params
-        module.fail_json(msg=msg)
+    module.exit_json(**result)
```

Comment:
This changes the existing behaviour of module exit. Please add porting guide [here](https://docs.ansible.com/ansible/devel/porting_guides/porting_guide_2.8.html#noteworthy-module-changes) and add a change log entry [here](https://github.com/ansible/ansible/tree/devel/changelogs/fragments).

Reply:
Also, add test case for this change.

Reply:
Thamls @Akasurde for the review, I added a changelog fragment and there is already a test-case in https://github.com/ansible/ansible/pull/54882 for this use-case.